core/src/main/java/org/elasticsearch/index/fielddata/plain/GeoPointArrayAtomicFieldData.java

```
@@ -0,0 +1,147 @@
+import org.elasticsearch.index.fielddata.MultiGeoPointValues;
+import org.elasticsearch.index.fielddata.ordinals.Ordinals;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ *
+ */
+public abstract class GeoPointArrayAtomicFieldData extends AbstractAtomicGeoPointFieldData {
+    @Override
+    public void close() {
+    }
+
+    static class WithOrdinals extends GeoPointArrayAtomicFieldData {
+        private final LongArray indexedPoints;
+        private final Ordinals ordinals;
+        private final int maxDoc;
+
+        public WithOrdinals(LongArray indexedPoints, Ordinals ordinals, int maxDoc) {
+            super();
+            this.indexedPoints = indexedPoints;
+            this.ordinals = ordinals;
+            this.maxDoc = maxDoc;
+        }
+
+        @Override
+        public long ramBytesUsed() {
+            return RamUsageEstimator.NUM_BYTES_INT + indexedPoints.ramBytesUsed();
+        }
+
+        @Override
+        public Collection<Accountable> getChildResources() {
+            List<Accountable> resources = new ArrayList<>();
+            resources.add(Accountables.namedAccountable("indexedPoints", indexedPoints));
+            return Collections.unmodifiableList(resources);
+        }
+
+        @Override
+        public MultiGeoPointValues getGeoPointValues() {
+            final RandomAccessOrds ords = ordinals.ordinals();
+            final SortedDocValues singleOrds = DocValues.unwrapSingleton(ords);
+            final GeoPoint point = new GeoPoint();
+            if (singleOrds != null) {
+                final GeoPointValues values = new GeoPointValues() {
+                    @Override
+                    public GeoPoint get(int docID) {
+                        final int ord = singleOrds.getOrd(docID);
+                        if (ord >= 0) {
+                            point.resetFromIndexHash(indexedPoints.get(ord));
+                        }
```

This is a little spooky: should we throw an exception in the else clause here?  Else the get of a `docId` that didn't index a geo point will silently return whatever `point` previously had?